purchase_order_qweb_report_adj/reports/purchase_order_templates.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template
+        id="report_purchaseorder_document_adj"
+        inherit_id="purchase.report_purchaseorder_document"
+        priority="100"
+    >
+        <xpath expr="//div[@t-if='o.partner_ref']" position="attributes">
+            <attribute name="t-att-style">'display: none'</attribute>
+        </xpath>
+        <xpath expr="//div[@t-if='o.partner_ref']" position="after">
+            <div t-if="o.partner_ref" class="col-3 bm-2">
+                <strong>Your Order Reference:</strong>
+                <p t-field="o.partner_ref" class="m-0" />
+            </div>
```

Comment:
Can you try adding the translation in `i18n_extra` folder instead of doing this?

Copy `ja.po` of the purchase module, remove unnecessary parts, and adjust the translation of `<strong>Your Order Reference:</strong>`.